### DISCUSSION TITLE
Connecting to a remote redis

### DISCUSSION THREAD

**Discussion by jerometremblay** [sidekiq/sidekiq#6013] [Q&A]
I am running a sidekiq process on AWS, and my REDIS server is elsewhere.

I added an initializer like this

```
# config/initializers/sidekiq.rb
Sidekiq.configure_server do |config|
  config.redis = { url: 'rediss://username:password@host.com:6379' }
end
Sidekiq.configure_client do |config|
  config.redis = { url: 'rediss://username:password@host.com:6379' }
end
```

But when Sidekiq starts, I get the `NOAUTH Authentication required.` error.

When I try to debug it, i can do

```
redis = Redis.new(url: 'rediss://username:password@host.com:6379')
redis.ping
```
and receive my pong.

I tried configuring username, host, password individually instead of using the URL, but it doesn't help.

What am I doing wrong?

```
    sidekiq (7.1.2)
      concurrent-ruby (< 2)
      connection_pool (>= 2.3.0)
      rack (>= 2.2.4)
      redis-client (>= 0.14.0)

```

**Comment by mperham** [comment]
Maybe you are trying to contact an older Redis with RESP2 when Sidekiq 7 requires RESP3?

**Reply by jerometremblay** [reply]
Using redis-cli I confirmed that I run redis 6.2.5.

```
host.com:6379> info
# Server
redis_version:6.2.5
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:19d2f4c94e0b6820
redis_mode:standalone
os:Linux 5.19.0-1029-aws x86_64
```

According to [the documentation](https://redis.io/docs/reference/protocol-spec/), RESP3 is opt-in, but I could not figure out if it is enabled or not in my Sidekiq settings or how to enable it.

**Reply by jerometremblay** [reply]
I can use HELLO 2 and HELLO 3 in redis-cli, so the server definitely supports it.

**Reply by mperham** [reply]
Is this a Redis Labs instance? They report 6.2 but only the new 7.2 tier supports Sidekiq 7.

**Reply by jerometremblay** [reply]
I don't know. I don't have a direct access to the redis server.

From the redis_build_id I found [a reference](https://fr.wikibooks.org/wiki/Le_syst%C3%A8me_d%27exploitation_GNU-Linux/Redis) that hints about the freely available version.
